torch/distributed/distributed_c10d.py

```diff
@@ -588,22 +597,36 @@ def _new_process_group_helper(world_size,
         prefix_store = PrefixStore(group_name, store)
 
         if backend == Backend.GLOO:
+            if pg_options is None:
+                pg_options = ProcessGroupGloo.Options()
+                pg_options.timeout = timeout
+                ifname_env = getenv("GLOO_SOCKET_IFNAME")
+                if ifname_env is not None:
+                    pg_options.devices = [ProcessGroupGloo.create_device(interface=iface) for iface in ifname_env.split(",")]
+                else:
+                    pg_options.devices = [ProcessGroupGloo.create_default_device()]
+
             pg = ProcessGroupGloo(
                 prefix_store,
                 rank,
                 world_size,
-                timeout=timeout)
+                pg_options)
             _pg_map[pg] = (Backend.GLOO, store)
             _pg_names[pg] = group_name
         elif backend == Backend.NCCL:
             if not is_nccl_available():
                 raise RuntimeError("Distributed package doesn't have NCCL "
                                    "built in")
+            if pg_options is None:
+                pg_options = ProcessGroupNCCL.Options()
+                pg_options.is_high_priority_stream = False
+                pg_options.timeout = timeout
+
             pg = ProcessGroupNCCL(
                 prefix_store,
                 rank,
                 world_size,
-                timeout)
+                pg_options)
```

Comment:
we should validate that this is an instance of `ProcessGroupNCCLOptions`, more generally should we validate that the options struct matches the backend and raise a useful error if it doesn't?